test/testSamples/MaterialX/multiple_connections.usda

```diff
@@ -101,5 +101,65 @@ def Material "Material1" (
             uniform float2 ui:nodegraph:node:pos = (-1.1293055, -3.50525)
         }
     }
+
```

Comment:
Should you also have a combine with no separate case?

Reply:
And a separate with no combine case?

Reply:
The code does not use any knowledge except the id strings and is very generic in how it traverses inputs. It will work across exotic scenarios (like a combine with one connection from a separate and another from a float node) without having to test every single permutation.